rust/fory-derive/src/util.rs

```diff
@@ -17,38 +17,73 @@
 
 use syn::{Field, Fields, GenericArgument, PathArguments, Type, TypePath, TypeTraitObject};
 
-pub fn sorted_fields(fields: &Fields) -> Vec<&Field> {
-    let fields = fields.iter().collect::<Vec<&Field>>();
+/// Sorted field with its original index preserved.
+///
+/// For tuple structs, `original_index` is the field's position in the original
+/// struct definition (0, 1, 2, ...), which is used as the field name.
+#[derive(Clone, Copy)]
+pub struct SortedField<'a> {
```

Comment:
This name looks a little vargue to me. the sort is the semantics of a collection, the element itself does not have this semantics.

How about just name as `StructField`?

Reply:
And `extract_fields` will make more sense

Reply:
> What I want to express is fields with original index, how about `IndexedField`?

How about `IndexedField`?

Reply:
This cames to my mind at first. But we may add more info to this struct, `IndexedField` may looks vargue in the future.